config/user_data.go

```diff
@@ -2,38 +2,38 @@ package config
 
 import (
 	"fmt"
-
-	"gopkg.in/yaml.v3"
+	"time"
 
 	"github.com/buildwithgrove/path/config/utils"
 )
 
+const defaultCacheRefreshInterval = 5 * time.Minute
+
 /* --------------------------------- User Data Config Struct -------------------------------- */
 
-// UserDataConfig contains user data configuration settings.
+// UserDataConfig contains user data configuration settings. This is off by default.
+//
+// If the 'user_data_config' field in the config YAML is set, the related features will become active.
```

Comment:
See above comment. Database requirement is strictly tied to user data functionality being enabled.

In the near future when we added throughput limiting there will be a requirement for other deps (eg. Redis, Gubernator sidecar, etc.) which are also under the user data umbrella. I think splitting all those deps out in their own config will be far more confusing to users than enforcing that everything required for user data lives under one umbrella term in the config.

Reply:
Agreed, but I'm thinking from an outsider's perspective.

User: "How do I configure the database?"'
Us: "It's under user config"
User: "Why?"
Us: "Because the database is only used for user data"
User: "Oh okay, I get it".

_* Some time later *_

New user: "How do I configure the database?"
_Repeat conversation

_* Some time later *_

Us #1: "We need to reuse the database for something other than user"
Us #2: "Okay, do we need to split out the config?"
Us #1: "Probably, but that's a lot of changes right now"

Reply:
Could you provide a quick tl;dr here for context? (Lots of different thread on different topics)

Reply:
tl;dr is with the Envoy Auth Plugin all authorization of requests is performed in the Go plugin that runs alongside the Envoy Proxy. This means that the database connection happens there, PATH doesn't know or care about a database and doesn't need anything related to either the database or user data.

[The full Notion doc has far more detail about this](https://www.notion.so/buildwithgrove/Envoy-Auth-Plugin-103a36edfff680aeb231d8bb01b4d4b8) but that's a quick summary.

Reply:
What do you think about just removing the user data/database from the config altogether since it's likely we will be moving the whole `db` package into the auth plugin. If something causes us not to take this approach I can add it back later with an improved name, though I think that's not likely to be necessary.

Reply:
I've removed everything related to the database from the config.